### DISCUSSION TITLE
`ls` grid view (AKA `lsg`) with optional argument and `-a` flag

### DISCUSSION THREAD

**Discussion by J-Kappes** [nushell/nushell#10413] [Show and tell]
```nu
def lsg [
    path = ""
    --all (-a)
] {
  if $all {
    ls -as $path
  } else {
    ls -s  $path
  }
  | sort-by type name -i | grid -c
}
```
uses `ls -s` flag to display only filename, not path, as to not defeat the purpose of allowing grid view for `ls` output for arbitrary directories.

**Comment by fdncred** [comment]
Some of the LS_COLORS are lost with `ls -s` unfortunately. That's a long standing bug. What's the `str trim` about? Is `grid -c` returning something it shouldn't?

**Reply by J-Kappes** [reply]
>What's the str trim about?

Idk actually. I copied most of this command from somewhere (I think you posted it) and only added the argument.

**Reply by fdncred** [reply]
ya, `lsg` is my creation. I've used it for years now.
```
def lsg [] { ls | sort-by type name -i | grid -c }
```

**Reply by J-Kappes** [reply]
I think it was on some showcase of how users used custom commands and that specific implementation was meant to run every time you `cd`. I can't find it anymore but I'm pretty sure I didn't add the `str trim`.

**Reply by fdncred** [reply]
yup, i use it as a hook like this
```nushell
    env_change: {
      PWD: [
      # Print lsg on each folder change
      { |before, after|
        print (lsg)
      }
...
```
